### PR TITLE
timeutil: inline github.com/jeffjen/datefmt

### PR DESCRIPTION
**Pull request by tamird**
https://github.com/jeffjen/datefmt/pull/3 is needed to support windows
but the author has been slow to respond. Inlining for now (for ever?).

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13750)

### CODE REVIEW

**Comment by knz** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-:-KdfkZbAAYfHrsrCCN4g:bnfp4nl)*

**Comment by petermattis** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[glide.lock, line 404 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-KdfkI2cmwyiEkrcIgxO:-KdfkI2cmwyiEkrcIgxP:b-a2wvzd) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd3cdf2c65272ad72a7bc17acf91b174fed89cb5/glide.lock#L404)):*
> ```
>   - transport
> - name: gopkg.in/yaml.v2
>   version: a3f3340b5840cee44f372bddb5880fcbc419b46a
> ```

Can we leave out the changes to this package and c-protobuf? I'm sure they are innocuous, yet also unrelated to this change.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750)*

**Comment by petermattis** [comment]
:lgtm: (module the extraneous vendor bumps)

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-:-KdfkqVFTlplxBDwCBLf:b-ymigi4)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[glide.lock, line 404 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-KdfkI2cmwyiEkrcIgxO:-KdfkvpsQsc6VvJbuo5B:b-73v9yl) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd3cdf2c65272ad72a7bc17acf91b174fed89cb5/glide.lock#L404)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Can we leave out the changes to this package and c-protobuf? I'm sure they are innocuous, yet also unrelated to this change.
</blockquote></details>

This is actually reverting an earlier unintended change made in f70ad7ea6eaab3cebe3b30ac0df09e5d24fb4fab.

The c-protobuf change is moving to the merge commit (previously was at that merge commit's right parent, which is identical).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750)*

**Comment by petermattis** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[glide.lock, line 404 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-KdfkI2cmwyiEkrcIgxO:-KdflaCRGrGRgneI2k3v:bsh3jwi) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd3cdf2c65272ad72a7bc17acf91b174fed89cb5/glide.lock#L404)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

This is actually reverting an earlier unintended change made in f70ad7ea6eaab3cebe3b30ac0df09e5d24fb4fab.

The c-protobuf change is moving to the merge commit (previously was at that merge commit's right parent, which is identical).
</blockquote></details>

Ok, still would be more obvious as a separate PR, but I suppose dealing with vendor is a sufficient headache to not warrant that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750)*

**Comment by bdarnell** [comment]
LGTM. Inlining this forever makes sense to me; it's not really doing anything but passing through to libc. 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[pkg/util/timeutil/conv.go, line 28 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-KdfnHYa3Bv7Y4jCn7Ya:-KdfnHYa3Bv7Y4jCn7Yb:b1olllk) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2e95c531d3c2043b24a8bacbe61bc91e48d604b/pkg/util/timeutil/conv.go#L28)):*
> ```Go
> // Strftime converts a time to a string using some C-style format.
> func Strftime(time time.Time, fmt string) (string, error) {
> 	return timeutil.Strftime(&time, fmt), nil
> ```

This unrelated change needs to be discussed separately. We had a platform-independent `strftime` implementation, and now we're proxying to the host platform's implementation (in particular, not all platforms support microseconds in `strftime`, according to #9762). This increases our exposure to issues like #10130 (if that's still an issue, this PR will probably introduce an analogous issue for `strftime`). 

I think I'm OK with this change (I'd prefer to use portable versions of both, but as long as we're using libc's `strptime` we might as well get `strftime` from the same place), but it needs more visibility than being tucked into this PR. CC @knz 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-:-KdfpVGNSMAl5icvXIHS:b-y1gis1)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[pkg/util/timeutil/conv.go, line 28 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-KdfnHYa3Bv7Y4jCn7Ya:-Kdfpu5lkPgOv26GobaM:b-jwo5t7) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2e95c531d3c2043b24a8bacbe61bc91e48d604b/pkg/util/timeutil/conv.go#L28)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

This unrelated change needs to be discussed separately. We had a platform-independent `strftime` implementation, and now we're proxying to the host platform's implementation (in particular, not all platforms support microseconds in `strftime`, according to #9762). This increases our exposure to issues like #10130 (if that's still an issue, this PR will probably introduce an analogous issue for `strftime`). 

I think I'm OK with this change (I'd prefer to use portable versions of both, but as long as we're using libc's `strptime` we might as well get `strftime` from the same place), but it needs more visibility than being tucked into this PR. CC @knz 
</blockquote></details>

Oh, I just missed this. This also explains why the tests are failing; I'll revert it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750)*

**Comment by dt** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[glide.lock, line 404 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-KdfkI2cmwyiEkrcIgxO:-KdfoIVeDKrgUZ13OpHR:bgc7p6h) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd3cdf2c65272ad72a7bc17acf91b174fed89cb5/glide.lock#L404)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Ok, still would be more obvious as a separate PR, but I suppose dealing with vendor is a sufficient headache to not warrant that.
</blockquote></details>

huh, I would have expected the pins to prevent unintended changes in #13714...  though it looks like this is actually just bringing the lockfile in line with the pin?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750)*

**Comment by tamird** [comment]
Review status: 1 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[glide.lock, line 404 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750#-KdfkI2cmwyiEkrcIgxO:-KdfrSu0NYJCVVbni6ao:b-eb7eu2) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd3cdf2c65272ad72a7bc17acf91b174fed89cb5/glide.lock#L404)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

huh, I would have expected the pins to prevent unintended changes in #13714...  though it looks like this is actually just bringing the lockfile in line with the pin?
</blockquote></details>

Yeah, it was my fault, I probably forgot to update vendor after some changes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13750)*

File: pkg/util/timeutil/conv.go [hunk 1]
```diff
@@ -18,20 +18,47 @@ package timeutil
...
-	"github.com/jeffjen/datefmt"
 	"github.com/leekchan/timeutil"
+	"github.com/pkg/errors"
 )
 
+// This is a modified version of github.com/jeffjen/datefmt inlined here until
+// https://github.com/jeffjen/datefmt/pull/3 is merged.
+
+// #include <stdlib.h>
+// #define __USE_XOPEN
```
Comment (tamird) [inline pkg/util/timeutil/conv.go]: for anyone following at home: yes this a thing

**Pull request merged**